src/kprobe.bpf.c

```diff
@@ -0,0 +1,27 @@
+// SPDX-License-Identifier: GPL-2.0 OR BSD-3-Clause
+/* Copyright (c) 2021 Sartura */
+#include "vmlinux.h"
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_tracing.h>
+#include <bpf/bpf_core_read.h>
+
+char LICENSE[] SEC("license") = "Dual BSD/GPL";
+
+SEC("kprobe/kernel_clone")
+int BPF_KPROBE(kernel_clone, struct kernel_clone_args *args)
+{
+	unsigned long stack_size, stack, tls;
```

Comment:
this is going to be a problem on 32-bit architectures. unsigned long is always 64-bit for BPF, even on 32-bit architectures. So you'll be reading wrong amount of bytes. It's handled by libbpf automatically for fentry/fexit programs with direct memory accesses, but not for BPF_CORE_READ. Reading longs from kernel is tricky. Is there any other good example you can use?

Reply:
How about tracking exec() and printing PID and filename of the process? Sort of mini-execsnoop.

Reply:
Switching to exec turned out to be somewhat more complex than I expected it to be, due to fentry/fexit issues. I'll describe the issues I encountered in a reply to your comment about fentry below. kprobe now tracks bprm_execve and prints the PID and filename. The README also reflects those changes